src/main/java/net/mcreator/ui/modgui/ParticleGUI.java

```diff
@@ -0,0 +1,225 @@
+
+import net.mcreator.blockly.data.Dependency;
+import net.mcreator.element.types.Particle;
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.MCreatorApplication;
+import net.mcreator.ui.component.SearchableComboBox;
+import net.mcreator.ui.component.util.ComboBoxUtil;
+import net.mcreator.ui.component.util.ComponentUtils;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.dialogs.TextureImportDialogs;
+import net.mcreator.ui.help.HelpUtils;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.init.UIRES;
+import net.mcreator.ui.laf.renderer.WTextureComboBoxRenderer;
+import net.mcreator.ui.minecraft.ProcedureSelector;
+import net.mcreator.ui.validation.AggregatedValidationResult;
+import net.mcreator.ui.validation.Validator;
+import net.mcreator.ui.validation.component.VComboBox;
+import net.mcreator.util.ListUtils;
+import net.mcreator.workspace.elements.ModElement;
+import net.mcreator.workspace.elements.VariableElementType;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+import java.io.File;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+public class ParticleGUI extends ModElementGUI<Particle> {
+
+	private final VComboBox<String> texture = new SearchableComboBox<>();
+
+	private final JSpinner width = new JSpinner(new SpinnerNumberModel(0.2, 0, 4096, 0.1));
+	private final JSpinner height = new JSpinner(new SpinnerNumberModel(0.2, 0, 4096, 0.1));
+	private final JSpinner scale = new JSpinner(new SpinnerNumberModel(1, 0.1, 4096, 0.1));
+	private final JSpinner gravity = new JSpinner(new SpinnerNumberModel(0, -100, 100, 0.1));
+	private final JSpinner speedFactor = new JSpinner(new SpinnerNumberModel(1, -100, 100, 0.1));
+	private final JSpinner maxAge = new JSpinner(new SpinnerNumberModel(7, 0, 100000, 1));
+	private final JSpinner maxAgeDiff = new JSpinner(new SpinnerNumberModel(0, 0, 100000, 1));
+
+	private final JCheckBox canCollide = L10N.checkbox("elementgui.common.enable");
+	private final JCheckBox alwaysShow = L10N.checkbox("elementgui.common.enable");
+	private final JCheckBox animate = L10N.checkbox("elementgui.common.enable");
+
+	private final JComboBox<String> renderType = new JComboBox<>(new String[] { "OPAQUE", "TRANSLUCENT", "LIT" });
+
+	private ProcedureSelector additionalExpiryCondition;
+
+	public ParticleGUI(MCreator mcreator, ModElement modElement, boolean editingMode) {
+		super(mcreator, modElement, editingMode);
+		this.initGUI();
+		super.finalizeGUI();
+	}
+
+	@Override protected void initGUI() {
+		additionalExpiryCondition = new ProcedureSelector(this.withEntry("particle/additional_expiry_condition"),
+				mcreator, L10N.t("elementgui.particle.expiry_condition"), ProcedureSelector.Side.CLIENT, true,
+				VariableElementType.LOGIC,
+				Dependency.fromString("x:number/y:number/z:number/world:world/age:number/onGround:logic"));

```

Comment:
onGround should be boolean, or it'll appear as a white square when used in a procedure

Reply:
Will fix it, thanks for the heads-up ;)